text/0497-glue-l2-construct.md

```diff
@@ -0,0 +1,434 @@
+    description: 'Scheduled run for ' + myGlueJob.name,
+    actions: [ myGlueCrawler]
+});
+
+// Create Weekly schedule at 00 UTC on Sunday
+myWorkflow.createWeeklyScheduleTrigger(this, 'TriggerJobOnWeeklySchedule', {
+    description: 'Scheduled run for ' + myGlueJob.name,
+    actions: [glueJob1, glueJob2, glueJob3, glueCrawler]
+});
+
+// Create Custom schedule, e.g. Monthly on the 7th day at 15:30 UTC
+myWorkflow.createCustomScheduleJobTrigger(this, 'TriggerCrawlerOnCustomSchedule', {
+    description: 'Scheduled run for ' + myGlueJob.name,
+    actions: [glueJob1, glueJob2, glueJob3, glueCrawler]
+    schedule: events.Schedule.cron(day: '7', hour: '15', minute: '30')
+});
+```
+
+#### **3. Notify  Event Triggers**
+
+This type of trigger is only supported with Glue workflows. There are two types
+of notify event triggers, batching and non-batching trigger. For batching triggers,
+developers must specify `BatchSize` but for non-batching `BatchSize` will be set
+to 1. For both triggers, `BatchWindow` will be default to 900 seconds.
+
+```
+myWorkflow.createNotifyEventTrigger(this, 'MyNotifyTriggerBatching', {
+    batchSize: int,
+    jobActions: [glueJob1, glueJob2, glueJob3],
+    actions: [glueJob1, glueJob2, glueJob3, glueCrawler]
+});
+
+myWorkflow.createNotifyEventTrigger(this, 'MyNotifyTriggerNonBatching', {
+    actions: [glueJob1, glueJob2, glueJob3]
+});
+```
+
+#### **4. Conditional Triggers**
+
+Conditional triggers have a predicate and actions associated with them.
+When the predicateCondition is true, the trigger actions will be executed.
+
+```
+// Triggers on Job and Crawler status
+myWorkflow.createConditionalTrigger(this, 'conditionalTrigger', {
+    description: 'Conditional trigger for ' + myGlueJob.name,
+    actions: [glueJob1, glueJob2, glueJob3, glueCrawler]
+    predicateCondition: glue.TriggerPredicateCondition.AND,
+    jobPredicates: [{'job': glueJobPred, 'state': glue.JobRunState.FAILED},
+                    {'job': glueJobPred1, 'state' : glue.JobRunState.SUCCEEDED}]
+});
+```
+
+### Connection Properties
+
+A `Connection` allows Glue jobs, crawlers and development endpoints to access
+certain types of data stores.
+
+***Secrets Management
+    **User needs to specify JDBC connection credentials in Secrets Manager and
+    provide the Secrets Manager Key name as a property to the Job connection
+    property.
+
+* **Networking - CDK determines the best fit subnet for Glue Connection
+configuration
+    **The current glue-alpha-module requires the developer to specify the
+    subnet of the Connection when it’s defined. The developer can still specify the
+    specific subnet they want to use, but no longer have to. This Glue L2 RFC will
+    allow developers to provide only a VPC and either a public or private subnet
+    selection. The L2 will then leverage the existing [EC2 Subnet Selection](https://docs.aws.amazon.com/cdk/api/v2/python/aws_cdk.aws_ec2/SubnetSelection.html)
```

Comment:
Nice! It should have been like this from the beginning.

Reply:
Agreed!  :)